src/osd/OSD.cc

```diff
@@ -6264,23 +6265,42 @@ int OSD::probe_smart_device(const char *device, int timeout, std::string *result
 
   int ret = smartctl.spawn();
   if (ret != 0) {
-    derr << "failed run smartctl: " << smartctl.err() << dendl;
-    return ret;
+    std::string err_msg = "failed to run smartctl: " + smartctl.err();
+    derr << err_msg << dendl;
+    *result = err_msg;
+    return -1;
   }
 
   bufferlist output;
   ret = output.read_fd(smartctl.get_stdout(), 100*1024);
   if (ret < 0) {
-    derr << "failed read from smartctl: " << cpp_strerror(-ret) << dendl;
+    std::string err_msg = "failed to read from smartctl: " + cpp_strerror(-ret);  
+    derr << err_msg << dendl;
+    *result = err_msg;
   } else {
-    ret = 0;
     *result = output.to_str();
```

Comment:
I think this part can remain the same too... if we fail to read the output that is not good.